### PR TITLE
Blog - Create Automated CI/CD Builds using GitHub Actions and DockerHub

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks

### CODE REVIEW

**Comment by msali123** [comment]
[Preview Link](https://643ffda032ff9d309e15d21b--ephemeral-narwhal-3ac3b2.netlify.app/blog/cicd-build-github-action-dockerhub/)
@balapriyac

File: blog/_posts/2023-05-18-cicd-build-github-action-dockerhub.md [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+author: Rose Chege
+
+internal-links:
+ - CI/CD
+ - Github Actions
+ - DockerHub
+ - Automation
+---
+
+[Continuous integration (CI) and continuous delivery (CD)](https://about.gitlab.com/topics/ci-cd/); (CI/CD) incorporate the operating principles used by application development teams to deliver the application code changes through built-in automation, testing, and collaboration workflows. It eliminates the manual development workflows to create automated processes that are predictable and repeatable.
```
Comment (balapriyac) [inline blog/_posts/2023-05-18-cicd-build-github-action-dockerhub.md:15]:
```suggestion
Continuous integration (CI) and continuous delivery (CD) incorporate the operating principles used by application development teams to deliver the application code changes through built-in automation, testing, and collaboration workflows. It eliminates the manual development workflows to create automated processes that are predictable and repeatable.
```

File: blog/_posts/2023-05-18-cicd-build-github-action-dockerhub.md [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+- Familiarity with YAML file.
+
+## GitHub Actions and DockerHub
+
+CI/CD approach creates a pipeline between integrations and delivery cycles. The pipeline's packaged application is easily integrated with different tools to check code and automatically deploy to production.
+
+[GitHub Actions](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions) is one of the popular CI/CD tools. It allows you to automate workflows through code builds, tests, and code deployment through [GitHub](https://github.com/) repositories. GitHub Actions can run tests, build Docker images and deploy applications to cloud infrastructures.
+
+[DockerHub](https://hub.docker.com/) is a [container registry](/blog/how-to-setup-and-use-amazons-elastic-container-registry) service for storing and distributing Docker images. A Docker image packages the application setting, dependencies, and codebase to create a portable [artifact](https://www.jetbrains.com/teamcity/ci-cd-guide/concepts/artifact-repository/#:~:text=Build%20artifacts%20are%20the%20files,available%20to%20your%20CI%20server.) that is easier to integrate between different CI/CD pipeline phases.
```
Comment (balapriyac) [inline blog/_posts/2023-05-18-cicd-build-github-action-dockerhub.md:38]:
```suggestion
[DockerHub](https://hub.docker.com/) is a [container registry](/blog/how-to-setup-and-use-amazons-elastic-container-registry) service for storing and distributing Docker images. A Docker image packages the application setting, dependencies, and codebase to create a portable artifact that is easier to integrate between different CI/CD pipeline phases.
```

File: blog/_posts/2023-05-18-cicd-build-github-action-dockerhub.md [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+
+GitHub actions require you to host your application code on a remote repository, including the `Dockerfile` containing the Docker commands. A GitHub repository store and manage your code and related files. Teams can push changes to this repository to update the application codebase collaboratively.
+
+[Create a GitHub repository](https://docs.github.com/en/get-started/quickstart/create-a-repo) to add all your code.
+
+Once you create the repository, add your application files by uploading them using [Git commands](https://git-scm.com/docs) or [GitHub Desktop](https://desktop.github.com/).
+
+For simplicity, I recommend using a Git client with a graphical user interface (GUI), such as GitHub Desktop, [Sourcetree](https://www.sourcetreeapp.com/), or [GitKraken](https://www.gitkraken.com/), to make working with Git repositories on GitHub easier. Here's a brief overview of how to use GitHub Desktop to push code changes to your remote GitHub repository:
+
+- Open [GitHub](/blog/ci-comparison) Desktop and log in using your GitHub account
```
Comment (balapriyac) [inline blog/_posts/2023-05-18-cicd-build-github-action-dockerhub.md:209]:
```suggestion
- Open GitHub Desktop and log in using your GitHub account
```

File: blog/_posts/2023-05-18-cicd-build-github-action-dockerhub.md [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+ - Github Actions
+ - DockerHub
+ - Automation
+---
+
+[Continuous integration (CI) and continuous delivery (CD)](https://about.gitlab.com/topics/ci-cd/); (CI/CD) incorporate the operating principles used by application development teams to deliver the application code changes through built-in automation, testing, and collaboration workflows. It eliminates the manual development workflows to create automated processes that are predictable and repeatable.
+
+This eliminates human interventions and reduces workflow errors while ensuring code quality. Teams can frequently implement small code changes that will be automatically tested and pushed for delivery and [deployment](/blog/deployment-strategies).
+
+You can streamline your workflow using [GitHub Action](https://github.com/features/actions) and automate CI/CD Builds. This guide will teach you how to leverage GitHub Action to automate Docker builds to DockerHub. You will build an automated CI/CD builds using GitHub Actions and DockerHub.
```
Comment (balapriyac) [inline blog/_posts/2023-05-18-cicd-build-github-action-dockerhub.md:19]:
```suggestion
You can streamline your workflow using GitHub Actions and automate CI/CD Builds. This guide will teach you how to leverage GitHub Action to automate Docker builds to DockerHub. You will build an automated CI/CD builds using GitHub Actions and DockerHub.
```

File: blog/_posts/2023-05-18-cicd-build-github-action-dockerhub.md (hunk 2 above)
Comment (balapriyac) [inline blog/_posts/2023-05-18-cicd-build-github-action-dockerhub.md:38]:
```suggestion
[DockerHub](https://hub.docker.com/) is a container registry service for storing and distributing Docker images. A Docker image packages the application setting, dependencies, and codebase to create a portable [artifact](https://www.jetbrains.com/teamcity/ci-cd-guide/concepts/artifact-repository/#:~:text=Build%20artifacts%20are%20the%20files,available%20to%20your%20CI%20server.) that is easier to integrate between different CI/CD pipeline phases.
```

**Review by balapriyac — CHANGES_REQUESTED** [review]
Thanks, Sameer

File: blog/_posts/2023-05-18-cicd-build-github-action-dockerhub.md [hunk 5]
```diff
@@ -0,0 +1,467 @@
+---
+title: "Create Automated CI/CD Builds using GitHub Actions and DockerHub"
```
Comment (balapriyac) [inline blog/_posts/2023-05-18-cicd-build-github-action-dockerhub.md:2]:
```suggestion
title: "Create Automated CI/CD Builds Using GitHub Actions and DockerHub"
```

**Review by balapriyac — CHANGES_REQUESTED** [review]

**Comment by msali123** [comment]
@balapriyac [Changes done](https://644119174a3a2c11b8987912--ephemeral-narwhal-3ac3b2.netlify.app/blog/cicd-build-github-action-dockerhub/)

**Comment by balapriyac** [comment]
@jalletto this is ready to merge.

**Review by jalletto — APPROVED** [review]

**Pull request merged**